### PR TITLE
[Behat] Changing my account password with token I received scenario

### LINKED ISSUE #9731: [Behat] Missing test to reset password through token
**Sylius version affected**: master

**Description**  
Maybe I'm wrong but but I don't found any behat feature to change password through token sent into resetting password email.

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Branch?         | 1.2
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no/yes
| Deprecations?   | no/yes 
| Related tickets | fixes #9731 
| License         | MIT

I've changed a behat page and its interface cause I've renamed an existing one... If you have a better idea, I'm opened. But the old name was confusing with missing one.

### CODE REVIEW

File: src/Sylius/Behat/Page/Shop/Account/RequestPasswordRequestResetPage.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Page\Shop\Account;
+
+use Behat\Mink\Exception\ElementNotFoundException;
+use Sylius\Behat\Page\SymfonyPage;
+
+class RequestPasswordRequestResetPage extends SymfonyPage implements RequestPasswordResetPageInterface
```
[the remark this answers is not in the document]
Comment (loic425) [reply src/Sylius/Behat/Page/Shop/Account/RequestPasswordRequestResetPage.php]: Lol yeah, it's a mistake :)

File: src/Sylius/Behat/Context/Setup/UserContext.php [hunk 2]
```diff
@@ -150,4 +158,18 @@ private function prepareUserVerification(UserInterface $user)
...
         $this->userManager->flush();
     }
+
+    /**
+     * @param UserInterface $user
+     */
+    private function prepareUserPasswordResetToken(UserInterface $user)
+    {
+        $token = 'itotallyforgotmypassword';
+        $this->sharedStorage->set('password_reset_token', $token);
```
[the remark this answers is not in the document]
Comment (loic425) [reply src/Sylius/Behat/Context/Setup/UserContext.php]: Do we keep this following code with storage on setup context ?
```php
$this->sharedStorage->set('password_reset_token', $token);
```

[the remark this answers is not in the document]
Comment (pamil) [reply src/Sylius/Behat/Context/Setup/UserContext.php]: I think it's no longer needed then.

**Comment by pamil** [comment]
Thank you, Loic! 🥇 🎉

**Pull request merged**